x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -165,7 +166,8 @@ private static boolean nodeSupportsModelSnapshotVersion(DiscoveryNode node, Job
             // which is OK as we have already checked the node is >= 5.5.0.
             return true;
         }
-        return node.getVersion().onOrAfter(job.getModelSnapshotMinVersion());
+        return MlConfigVersion.fromString(node.getAttributes().get(MachineLearning.ML_CONFIG_VERSION_NODE_ATTR))
+            .onOrAfter(job.getModelSnapshotMinVersion());
```

Comment:
Instead of `MlConfigVersion.fromString(node.getAttributes().get(MachineLearning.ML_CONFIG_VERSION_NODE_ATTR))` here you should write a helper method in `MlConfigVersion`. Something like `getMlConfigVersionForNode(DiscoveryNode, String)`. That can then get the attribute and convert it to an `MlConfigVersion`, but also account for the case of a node that's too old to have the attribute (by looking at its `Version`).

In fact, that helper method could then be used in the loop in `getMinMaxMlConfigVersion`.